sale_order_show_purchase_order/models/sale_order_line.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+
+
+from odoo import api, fields, models
+
+ 
+class SaleOrder(models.Model):
+
+    _inherit = 'sale.order'
+            
+    @api.multi
+    def _get_purchase_ids(self):       
+
+        group_list = []
+        purchase_list = []
+
+        purchases = self.purchase_order_ids.search([])
+
+        for record in self.order_line:
+            if record != False:
+                for line in self.order_line:
+                    if line.procurement_ids.ids != []:
+                        group_list.append(line.procurement_ids.group_id)
+
+            if group_list != []:
+                for group in group_list:
+                    print group
+                    for id_number in group.procurement_ids:
+                        if id_number.purchase_id:
```

Comment:
Check for `ids` before trying to access `ids[0]`
`if id_number.purchase_id and if id_number.purchase_id.ids`:

Reply:
Changed the condition to 
"if id_number.purchase_id and id_number.purchase_id.ids:"
This should work too?